src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -127,30 +143,34 @@ private async Task UpdateInstalledVersionsAsync(CancellationToken cancellationTo
             {
                 try
                 {
-                    IPackageReferenceContextInfo installedVersion = await GetInstalledPackageAsync(project.NuGetProject, Id, cancellationToken);
-                    if (installedVersion != null)
+                    (IPackageReferenceContextInfo TopLevelPackage, IPackageReferenceContextInfo TransitivePackage) installedPackage = await GetInstalledAndTransitivePackagesAsync(project.NuGetProject, Id, cancellationToken);
+                    var topLevelPackageVersion = installedPackage.TopLevelPackage;
+                    var transitivePackageVersion = installedPackage.TransitivePackage;
+
+                    project.InstalledVersionMaxVulnerability = -1;
+                    project.RequestedVersion = null;
+                    project.AutoReferenced = false;
+
+                    if (transitivePackageVersion != null)
+                    {
+                        project.PackageLevel = PackageLevel.Transitive;
+                        UpdateProjectInstallationInfo(project, transitivePackageVersion, installedVersionsSet);
+                    }
+                    else if (topLevelPackageVersion != null)
                     {
-                        project.InstalledVersion = installedVersion.Identity.Version;
-                        installedVersionsSet.Add(installedVersion.Identity.Version);
-                        project.AutoReferenced = installedVersion.IsAutoReferenced;
-
-                        if (project.NuGetProject.ProjectStyle.Equals(ProjectStyle.PackageReference))
-                        {
-                            project.RequestedVersion = installedVersion?.AllowedVersions?.OriginalString;
-                        }
-
-                        if (_searchResultPackage.VulnerableVersions.TryGetValue(installedVersion.Identity.Version, out int vulnerable))
-                        {
-                            project.InstalledVersionMaxVulnerability = vulnerable;
-                            vulnerabilitiesSet.Add(installedVersion.Identity.Version);
-                        }
+                        project.PackageLevel = PackageLevel.TopLevel;
+                        UpdateProjectInstallationInfo(project, topLevelPackageVersion, installedVersionsSet);
                     }
                     else
                     {
-                        project.RequestedVersion = null;
                         project.InstalledVersion = null;
-                        project.AutoReferenced = false;
-                        project.InstalledVersionMaxVulnerability = -1;
+                        project.PackageLevel = null;
```

Comment:
Nit: In a previous comment I suggested moving these defaults back to the else case, but if there are some defaults that must be set for either top-level or transitive, then maybe just keep the defaults all together above. It's up to you.